### PR TITLE
Port System.ComponentModel new docs (second pass)

### PR DESCRIPTION
**Pull request by carlossanlop**
Ports some missed in the first pass.

Contributes to https://github.com/dotnet/runtime/issues/105983

### CODE REVIEW

**Comment by carlossanlop** [comment]
@buyaa-n @steveharter can you please give me a code review?

**Comment by carlossanlop** [comment]
Ping @gewarren and @dotnet/area-system-componentmodel , can I please get a review?

File: xml/System.ComponentModel/ComponentResourceManager.xml [hunk 1]
```diff
@@ -352,7 +352,13 @@
         <param name="value">To be added.</param>
         <param name="objectName">To be added.</param>
         <param name="culture">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>
+          <para>This method examines all the resources for the provided culture.</para>
```
Comment (gewarren) [inline xml/System.ComponentModel/ComponentResourceManager.xml:356]:
```suggestion
          <para>Examines all the resources for the provided culture.</para>
```

File: xml/System.ComponentModel/CustomTypeDescriptor.xml [hunk 2]
```diff
@@ -486,7 +486,10 @@
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
+        <summary>
+          <para>The GetConverterFromRegisteredType method returns a type converter for the type this type</para>
```
Comment (gewarren) [inline xml/System.ComponentModel/CustomTypeDescriptor.xml:490]:
```suggestion
          <para>Returns a type converter for the type this type</para>
```

File: xml/System.ComponentModel/CustomTypeDescriptor.xml [hunk 3]
```diff
@@ -1068,7 +1074,10 @@
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
+        <summary>
+          <para>The GetProperties method returns a collection of property descriptors</para>
```
Comment (gewarren) [inline xml/System.ComponentModel/CustomTypeDescriptor.xml:1078]:
```suggestion
          <para>Returns a collection of property descriptors</para>
```

File: xml/System.ComponentModel/CustomTypeDescriptor.xml [hunk 4]
```diff
@@ -1161,9 +1170,20 @@
         <ReturnType>System.Nullable&lt;System.Boolean&gt;</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>Whether types are required to be registered through <see cref="M:System.ComponentModel.TypeDescriptionProvider.RegisterType``1" />.</summary>
```
Comment (gewarren) [inline xml/System.ComponentModel/CustomTypeDescriptor.xml:1173]:
```suggestion
        <summary>Gets a value that indicates whether types are required to be registered through <see cref="M:System.ComponentModel.TypeDescriptionProvider.RegisterType``1" />.</summary>
```

File: xml/System.ComponentModel/ICustomTypeDescriptor.xml [hunk 5]
```diff
@@ -952,9 +952,23 @@ pd.GetValue(GetPropertyOwner(myPd));
         <ReturnType>System.Nullable&lt;System.Boolean&gt;</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>Whether types are required to be registered through <see cref="M:System.ComponentModel.TypeDescriptionProvider.RegisterType``1" />.</summary>
```
Comment (gewarren) [inline xml/System.ComponentModel/ICustomTypeDescriptor.xml:955]:
```suggestion
        <summary>Gets a value that indicates whether types are required to be registered through <see cref="M:System.ComponentModel.TypeDescriptionProvider.RegisterType``1" />.</summary>
```

File: xml/System.ComponentModel/TypeDescriptionProvider.xml [hunk 6]
```diff
@@ -438,7 +438,20 @@
       </Parameters>
       <Docs>
         <param name="instance">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>
+          <para>This method returns an extended custom type descriptor for the given object.</para>
```
Comment (gewarren) [inline xml/System.ComponentModel/TypeDescriptionProvider.xml:442]:
```suggestion
          <para>Returns an extended custom type descriptor for the given object.</para></summary>
```

File: xml/System.ComponentModel/TypeDescriptionProvider.xml [hunk 7]
```diff
@@ -438,7 +438,20 @@
...
+          <para>An extended type descriptor is a custom type descriptor that offers properties</para>
```
Comment (gewarren) [inline xml/System.ComponentModel/TypeDescriptionProvider.xml:443]: The remaining summary lines should go in remarks.

File: xml/System.ComponentModel/TypeDescriptionProvider.xml [hunk 8]
```diff
@@ -1036,7 +1049,14 @@
       </Parameters>
       <Docs>
         <param name="instance">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>
+          <para>This method returns a custom type descriptor for the given type / object.</para>
```
Comment (gewarren) [inline xml/System.ComponentModel/TypeDescriptionProvider.xml:1053]:
```suggestion
          <para>Returns a custom type descriptor for the given type or object.</para></summary>
```

File: xml/System.ComponentModel/TypeDescriptionProvider.xml [hunk 9]
```diff
@@ -1036,7 +1049,14 @@
...
+          <para>The objectType parameter is always valid, but the instance parameter may</para>
```
Comment (gewarren) [inline xml/System.ComponentModel/TypeDescriptionProvider.xml:1054]: Remaining lines should go in remarks.

File: xml/System.ComponentModel/TypeDescriptionProvider.xml [hunk 10]
```diff
@@ -1073,7 +1093,14 @@
       </Parameters>
       <Docs>
         <param name="objectType">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>
+          <para>This method returns a custom type descriptor for the given type / object.</para>
```
Comment (gewarren) [inline xml/System.ComponentModel/TypeDescriptionProvider.xml:1097]:
```suggestion
          <para>Returns a custom type descriptor for the given type or object.</para></summary>
```

File: xml/System.ComponentModel/TypeDescriptionProvider.xml [hunk 11]
```diff
@@ -1073,7 +1093,14 @@
...
+          <para>The objectType parameter is always valid, but the instance parameter may</para>
```
Comment (gewarren) [inline xml/System.ComponentModel/TypeDescriptionProvider.xml:1098]: Remaining lines should go in remarks.

File: xml/System.ComponentModel/TypeDescriptionProvider.xml [hunk 12]
```diff
@@ -1119,7 +1146,19 @@
       <Docs>
         <param name="objectType">To be added.</param>
         <param name="instance">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>
+          <para>This method returns a custom type descriptor for the given type / object.</para>
```
Comment (gewarren) [inline xml/System.ComponentModel/TypeDescriptionProvider.xml:1150]:
```suggestion
          <para>Returns a custom type descriptor for the given type / object.</para></summary>
```

File: xml/System.ComponentModel/TypeDescriptionProvider.xml [hunk 13]
```diff
@@ -1119,7 +1146,19 @@
...
+          <para>The objectType parameter is always valid, but the instance parameter may</para>
```
Comment (gewarren) [inline xml/System.ComponentModel/TypeDescriptionProvider.xml:1151]: Same comment as others.

File: xml/System.ComponentModel/TypeDescriptionProvider.xml [hunk 14]
```diff
@@ -1150,9 +1189,10 @@
       </Parameters>
       <Docs>
         <param name="type">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>Returns whether the type was registered with its provider through <see cref="M:System.ComponentModel.TypeDescriptor.RegisterType``1" />.</summary>
```
Comment (gewarren) [inline xml/System.ComponentModel/TypeDescriptionProvider.xml:1192]:
```suggestion
        <summary>Determines whether the type was registered with its provider through <see cref="M:System.ComponentModel.TypeDescriptor.RegisterType``1" />.</summary>
```

File: xml/System.ComponentModel/TypeDescriptionProvider.xml [hunk 15]
```diff
@@ -1150,9 +1189,10 @@
...
         <returns>To be added.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentNullException">If <paramref name="type" /> is <see langword="null" />.</exception>
```
Comment (gewarren) [inline xml/System.ComponentModel/TypeDescriptionProvider.xml:1195]:
```suggestion
        <exception cref="T:System.ArgumentNullException">
          <paramref name="type" /> is <see langword="null" />.</exception>
```

File: xml/System.ComponentModel/TypeDescriptionProvider.xml [hunk 16]
```diff
@@ -1267,7 +1307,11 @@
         <ReturnType>System.Nullable&lt;System.Boolean&gt;</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>
+          <para>Whether the provider uses reflection and requires types to be registered through <see cref="M:System.ComponentModel.TypeDescriptor.RegisterType``1" /></para>
```
Comment (gewarren) [inline xml/System.ComponentModel/TypeDescriptionProvider.xml:1311]:
```suggestion
          <para>Gets a value that indicates whether the provider uses reflection and requires types to be registered through <see cref="M:System.ComponentModel.TypeDescriptor.RegisterType``1" /></para>
```

File: xml/System.ComponentModel/TypeDescriptionProvider.xml [hunk 17]
```diff
@@ -1267,7 +1307,11 @@
...
+          <para>to support trimmed applications. For backwards compatibility, this is typically only enforced when calling members that have</para>
```
Comment (gewarren) [inline xml/System.ComponentModel/TypeDescriptionProvider.xml:1312]: Second sentence should go in remarks.

**Review by gewarren — APPROVED** [review]

File: xml/System.ComponentModel/TypeDescriptionProvider.xml [hunk 18]
```diff
@@ -1036,7 +1049,14 @@
...
+        <summary>
+          <para>This method returns a custom type descriptor for the given type / object.</para>
+          <para>The objectType parameter is always valid, but the instance parameter may</para>
+          <para>be <see langword="null" /> if no instance was passed to TypeDescriptor. The method should</para>
+          <para>return a custom type descriptor for the object. If the method is not</para>
+          <para>interested in providing type information for the object it should</para>
+          <para>return base.</para>
+        </summary>
         <returns>To be added.</returns>
         <remarks>To be added.</remarks>
```
Comment (carlossanlop) [inline xml/System.ComponentModel/TypeDescriptionProvider.xml:1061]:
```suggestion
        <summary>Returns a custom type descriptor for the given type / object.</summary>
        <returns>To be added.</returns>
        <remarks>The objectType parameter is always valid, but the instance parameter may be <see langword="null" /> if no instance was passed to TypeDescriptor. The method should return a custom type descriptor for the object. If the method is not interested in providing type information for the object it should return base.</remarks>
```

File: xml/System.ComponentModel/TypeDescriptionProvider.xml [hunk 19]
```diff
@@ -1073,7 +1093,14 @@
...
```
Comment (carlossanlop) [inline xml/System.ComponentModel/TypeDescriptionProvider.xml:1105]:
```suggestion
        <summary>Returns a custom type descriptor for the given type or object.</summary>
        <returns>To be added.</returns>
        <remarks>The objectType parameter is always valid, but the instance parameter may be <see langword="null" /> if no instance was passed to TypeDescriptor. The method should return a custom type descriptor for the object. If the method is not interested in providing type information for the object it should return base.</remarks>
```

File: xml/System.ComponentModel/TypeDescriptionProvider.xml [hunk 20]
```diff
@@ -1119,7 +1146,19 @@
...
+          <para>interested in providing type information for the object it should</para>
+          <para>return base.</para>
+          <para>This method is prototyped as virtual, and by default returns a</para>
+          <para>custom type descriptor that returns empty collections for all values</para>
+          <para>if no parent provider was passed. If a parent provider was passed,</para>
+          <para>this method will invoke the parent provider's GetTypeDescriptorFromRegisteredType</para>
+          <para>method.</para>
+        </summary>
         <returns>To be added.</returns>
         <remarks>To be added.</remarks>
```
Comment (carlossanlop) [inline xml/System.ComponentModel/TypeDescriptionProvider.xml:1163]:
```suggestion
        <summary>Returns a custom type descriptor for the given type or object.</summary>
        <returns>To be added.</returns>
        <remarks>
          <para>The objectType parameter is always valid, but the instance parameter may be <see langword="null" /> if no instance was passed to TypeDescriptor. The method should return a custom type descriptor for the object. If the method is not interested in providing type information for the object it should return base.</para>
          <para>This method is prototyped as virtual, and by default returns a custom type descriptor that returns empty collections for all values if no parent provider was passed. If a parent provider was passed, this method will invoke the parent provider's GetTypeDescriptorFromRegisteredType method.</para>
        </remarks>
```

File: xml/System.ComponentModel/TypeDescriptionProvider.xml [hunk 21]
```diff
@@ -1267,7 +1307,11 @@
...
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>
+          <para>Whether the provider uses reflection and requires types to be registered through <see cref="M:System.ComponentModel.TypeDescriptor.RegisterType``1" /></para>
+          <para>to support trimmed applications. For backwards compatibility, this is typically only enforced when calling members that have</para>
+          <para>a "FromRegisteredType" suffix.</para>
+        </summary>
         <value>To be added.</value>
         <remarks>To be added.</remarks>
```
Comment (carlossanlop) [inline xml/System.ComponentModel/TypeDescriptionProvider.xml:1316]:
```suggestion
        <summary>Gets a value that indicates whether the provider uses reflection and requires types to be registered through <see cref="M:System.ComponentModel.TypeDescriptor.RegisterType``1" /> to support trimmed applications.</summary>
        <value>To be added.</value>
        <remarks>For backwards compatibility, this is typically only enforced when calling members that have  "FromRegisteredType" suffix.</remarks>
```

File: xml/System.ComponentModel/TypeDescriptionProvider.xml [hunk 22]
```diff
@@ -438,7 +438,20 @@
...
+          <para>objects that reside in the same logical container. The GetTypeDescriptor</para>
+          <para>method does not return a type descriptor that provides these extra extended</para>
+          <para>properties. GetExtendedTypeDescriptor returns the set of these extended</para>
+          <para>properties. TypeDescriptor will automatically merge the results of these</para>
+          <para>two property collections. Note that while the .NET Framework component</para>
+          <para>model only supports extended properties this API can be used for extended</para>
+          <para>attributes and events as well, if the type description provider supports it.</para>
+        </summary>
         <returns>To be added.</returns>
         <remarks>To be added.</remarks>
```
Comment (carlossanlop) [inline xml/System.ComponentModel/TypeDescriptionProvider.xml:456]:
```suggestion
        <summary>Returns an extended custom type descriptor for the given object.</summary>
        <returns>To be added.</returns>
        <remarks>An extended type descriptor is a custom type descriptor that offers properties that other objects have added to this object, but are not actually defined on the object. For example, in the .NET Framework Component Model, objects that implement the interface IExtenderProvider can "attach" properties to other objects that reside in the same logical container. The GetTypeDescriptor method does not return a type descriptor that provides these extra extended properties. GetExtendedTypeDescriptor returns the set of these extended properties. TypeDescriptor will automatically merge the results of these two property collections. Note that while the .NET Framework component model only supports extended properties this API can be used for extended attributes and events as well, if the type description provider supports it.</remarks>
```

**Pull request merged**